<!-- -->
<launch>
  <arg name="is_linear_interpolation" default="True"/>
  <arg name="publishes_for_steering_robot" default="False"/>
  <arg name="add_virtual_end_waypoints" default="False"/>
  <arg name="const_lookahead_distance" default="4.0"/>
  <arg name="const_velocity" default="5.0"/>
  <arg name="lookahead_ratio" default="2.0"/>
  <arg name="minimum_lookahead_distance" default="6.0"/>

  <!-- 0 = waypoints, 1 = provided constant velocity -->
  <arg name="velocity_source" default="0"/>

  <!-- rosrun waypoint_follower pure_pursuit -->
  <node pkg="waypoint_follower" type="pure_pursuit" name="pure_pursuit" output="log">
    <param name="is_linear_interpolation" value="$(arg is_linear_interpolation)"/>
    <param name="publishes_for_steering_robot" value="$(arg publishes_for_steering_robot)"/>
    <param name="add_virtual_end_waypoints" value="$(arg add_virtual_end_waypoints)"/>
    <param name="const_lookahead_distance" value="$(arg const_lookahead_distance)"/>
    <param name="const_velocity" value="$(arg const_velocity)"/>
    <param name="lookahead_ratio" value="$(arg lookahead_ratio)"/>
    <param name="minimum_lookahead_distance" value="$(arg minimum_lookahead_distance)"/>
    <param name="velocity_source" value="$(arg velocity_source)"/>
  </node>
</launch>
